### PR TITLE
Add documentation to appconfig

### PR DESCRIPTION
**Pull request by maririos**

### CODE REVIEW

**Comment by dsgouda** [comment]
@maririos  Please pull from latest master and update the PR to fix the Travis failures

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs [hunk 1]
```diff
@@ -25,8 +34,8 @@ public ConfigurationSetting(string key, string value, string label = null)
         }
 
         /// <summary>
-        /// The primary identifier of a key-value.
-        /// The key is used in unison with the label to uniquely identify a key-value.
+        /// The primary identifier of a configuration setting.
```
Comment (pakrym) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs]: I've seen "Gets or sets " prefix being used for property documentation. We need to decide if we want to use it too.

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs [hunk 2]
```diff
@@ -12,16 +12,28 @@
...
+    /// <summary>
+    /// ConfigurationClient can be used for all operations with a <see cref="ConfigurationSetting"/> in the Azure Configuration Store.
+    /// </summary>
     public partial class ConfigurationClient
     {
         private readonly Uri _baseUri;
         private readonly HttpPipeline _pipeline;
 
+        /// <summary>
+        /// Creates a <see cref="ConfigurationClient"/> that sends requests to the configuration store.
```
Comment (pakrym) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs]: `Initializes a new instance of the ` is often use as ctor doc prefix in corefx

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs [hunk 3]
```diff
@@ -24,7 +30,7 @@ public class SettingSelector
         /// <remarks>See the documentation for this client library for details on the format of filter expressions</remarks>
         public IList<string> Labels { get; set; }
         /// <summary>
-        /// IKeyValue fields that will be retrieved.
+        /// Specify which fields will be retrieved.
```
Comment (jsquire) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs]: To me, the phrasing feels a bit awkward here.   Maybe we should consider something like `Controls the fields that will be populated` or `Allows requesting a specific set of fields` ?

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs (hunk 1 above)
Comment (jsquire) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs]: One vote against from me.   It's  the default template for properties and seems to get left a lot, but I question what value it provides.   Personally, I view the default case as being a read/write state and comment when that is not the case.

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs [hunk 4]
```diff
@@ -9,13 +9,22 @@
 
 namespace Azure.ApplicationModel.Configuration
 {
+    /// <summary>
+    /// ConfigurationSetting is a resource identified by unique combination of key and label.
```
Comment (jsquire) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs]: To me, the phrasing is a bit awkward.   Perhaps we should consider something like `A setting, defined by a unique combination of a key and label`.

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs [hunk 5]
```diff
@@ -12,6 +12,9 @@
 
 namespace Azure.ApplicationModel.Configuration
 {
+    /// <summary>
+    /// ConfigurationClient can be used for all operations with a <see cref="ConfigurationSetting"/> in the Azure Configuration Store.
```
Comment (jsquire) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationClient.cs]: Maybe we should consider something like `The client to use for interacting with the Azure Configuration Store` or something similar?   The name of the class leading the sentence feels a bit awkward to me.

**Review by jsquire — APPROVED** [review]
I'm assuming that we're intending the comments as the basis for consumers to use with intellisense and, potentially, generate documentation from.

As a nit, we've got some inconsistent use of punctuation across these, where some sentences end with a period and others do not.   We may want to consider making those consistent.

We've also got some mixed approaches to summaries, where some are more sentences-lie and the others more phrase-like.  We may want to consider a unified voice so that they're consistent across.  Personally, I believe that they make better documentation when they read as true sentences and offer a bit of context as well as the basic description of "what this thing does."

**Comment by maririos** [comment]
> I'm assuming that we're intending the comments as the basis for consumers to use with intellisense and, potentially, generate documentation from.

Correct

> As a nit, we've got some inconsistent use of punctuation across these, where some sentences end with a period and others do not. We may want to consider making those consistent.

Thanks! Modified

> We've also got some mixed approaches to summaries, where some are more sentences-lie and the others more phrase-like. We may want to consider a unified voice so that they're consistent across. Personally, I believe that they make better documentation when they read as true sentences and offer a bit of context as well as the basic description of "what this thing does."

I see. Modified

**Pull request merged**